### PR TITLE
PLANET-6579: Allow a drop down menu for the Donate button through the Navigation Bar

### PR DESCRIPTION
**Pull request by GP-Dan-Tovbein**
Ref: https://jira.greenpeace.org/browse/PLANET-6579

👉 [Parent PR](https://github.com/greenpeace/planet4-master-theme/pull/1628)

**How to test**
- First, check if the Donate menu has items already added going to `Appearence > Menus`
- Navigate to the [instance](https://www-dev.greenpeace.org/test-rhea/) and play around with the Donate menu.

Also go to:
`Planet4 > Navigation` and change the **Website Navigation Style** to `dark|light`

**What to test**
- Validate the Design iteration2_IA & nav MVP from [here](https://www.figma.com/file/Gn8wyqgto7608rlESePXZE/Foundations-%26-Components?node-id=0%3A103).
- Analytics vars set to the sub-menu items.

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [rhea](https://www-dev.greenpeace.org/test-rhea) | [admin](https://www-dev.greenpeace.org/test-rhea/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-rhea/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-rhea) | [composer-local.json](https://github.com/greenpeace/planet4-test-rhea/blob/main/composer-local.json)

:watch: 2022.02.16 13:08:40

File: src/MasterSite.php
Comment (mleray) [inline src/MasterSite.php:537]: I since found out that this function has a [second param for default value](https://github.com/greenpeace/planet4-master-theme/blob/master/functions.php#L73), so here instead of using `??` you could write `planet4_get_option( 'donate_button', '#' )`... Same on the following line 🙂

File: src/Settings.php
Comment (mleray) [inline src/Settings.php:657]: I came up with the copy for this on my own when working on the ticket, but probably someone (maybe @comzeradd?) should review it 😬

Comment (mleray) [reply src/Settings.php:657]: Maybe we also need documentation for this 🤔 It is a bit complicated since you need to create a new menu etc

Comment (mleray) [reply src/Settings.php:657]: Actually just saw that there is [documentation](https://planet4.greenpeace.org/manage/admin-setup/#donate-button) for the Donate button already, that will probably need to be updated?

File: assets/src/scss/layout/_navbar.scss
Comment (mleray) [inline assets/src/scss/layout/_navbar.scss:72]: I don't think we should use these variables for height or line-height, they are meant to be used for spacing values (so `margin`, `padding`, etc)

File: assets/src/scss/layout/_navbar.scss
Comment (mleray) [inline assets/src/scss/layout/_navbar.scss:326]: You probably want to use `margin-inline-end` here instead, so that it works for RTL sites as well

File: assets/src/scss/layout/_navbar.scss
Comment (GP-Dan-Tovbein) [reply assets/src/scss/layout/_navbar.scss:72]: Good point!

File: src/Settings.php
Comment (mleray) [inline src/Settings.php:675]: I mentioned it in the previous PR, so I'll say it here again 😁 I came up with this copy when I first worked on this, it should probably be reviewed by @comzeradd or Suzi!

[Our documentation](https://planet4.greenpeace.org/manage/admin-setup/#donate-button) for this section will also likely need an update, and we should add somewhere some directions for how to add a Donate button dropdown menu since it's a bit complicated.

File: assets/src/scss/layout/_navbar.scss
Comment (mleray) [inline assets/src/scss/layout/_navbar.scss:75]: This chevron is currently also showing in the mobile side menu, I don't think it should?

Screenshot 2022-02-25 at 15 01 37

File: src/Settings.php
Comment (GP-Dan-Tovbein) [reply src/Settings.php:675]: I agree! I left this suggestion into the description of the [base PR](https://github.com/greenpeace/planet4-master-theme/pull/1628) just to follow up before merge into master.

File: templates/navigation-submenu.twig
Comment (mleray) [inline templates/navigation-submenu.twig:4]: I think it would make more sense to call this parameter `menu` rather than `item` (so it would become `for key,item in menu.children`), the current notation is a bit confusing imo

Comment (GP-Dan-Tovbein) [reply templates/navigation-submenu.twig:4]: Completely agree!

File: templates/navigation-submenu.twig
Comment (mleray) [inline templates/navigation-submenu.twig:6]: I think we can remove these special cases already since we're dropping the Act/Explore navigation, and just use `item.title` everywhere

File: templates/navigation-submenu.twig
Comment (GP-Dan-Tovbein) [reply templates/navigation-submenu.twig:4]: @mleray I was working on this suggestion (initially agreed by me) but if we change the name to `menu` instead `item` each link inside menu will take `menu.*` values but it seems to be a bit inconsistent that, since it's not a menu, it's an link (or item). What do you think?

File: templates/navigation-submenu.twig
Comment (GP-Dan-Tovbein) [reply templates/navigation-submenu.twig:6]: Shouldn't they behave as a link? And what about Donate links?

Comment (mleray) [reply templates/navigation-submenu.twig:6]: They will still behave as links 🤔 This is just for the Google Analytics action parameter, and if I understood correctly with the new IA we no longer treat Act and Explore as special pages so here we should use `item.title` everywhere... Anyway, I see that we still use this in the navigation bar template so I guess we can keep this change for later in the process

Comment (GP-Dan-Tovbein) [reply templates/navigation-submenu.twig:6]: Sorry I miss explained 😶 for links I referred to `data-*` values. I asked about these values a week ago to Julia, so, is this [answer](https://jira.greenpeace.org/browse/PLANET-6509?focusedCommentId=245076&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-245076) solve your doubt?

File: assets/src/scss/layout/_navbar.scss
Comment (mleray) [inline assets/src/scss/layout/_navbar.scss:217]: Not really related to the Donate button menu but can be quickly solved, here it should be `flex-end` instead of `end` (which doesn't work in Safari)

Comment (mleray) [reply assets/src/scss/layout/_navbar.scss:217]: Actually it looks like `end` is [not even a valid value](https://css-tricks.com/almanac/properties/j/justify-content/) for this property, it's lucky that it works in some browsers 😄

File: templates/navigation-submenu.twig
Comment (mleray) [reply templates/navigation-submenu.twig:6]: Well not really, it looks like she agrees with me? Julia says to use `item.title`, she doesn't mention exceptions for the Act and Explore pages... But this is not related to the Donate button so it can be solved in another PR 🙃

File: templates/navigation-bar-new.twig
Comment (mleray) [inline templates/navigation-bar-new.twig:61]: Why do you use `only` here? 🤔 We don't use it anywhere else, including in the other [inclusion of this same template](https://github.com/greenpeace/planet4-master-theme/pull/1626/files#diff-3ad24a02d9e63bde53c8195a030484bfb200dac551677c03584386ba3394c92aR45)

File: assets/src/js/menu_editor.js
Comment (GP-Dan-Tovbein) [inline assets/src/js/menu_editor.js:160]: @lithrel I had to change the `map` to `filter`. Bacause `map` will always returns a value and in consequence if you try to get the zero index of the returned array the value will be always `null` in cases where you have more than 1 item into the list.
i.e. Having this array: `['navigation-bar-menu', 'donate-menu']`, then:

If you want to check `navigation-bar-menu` the result will be:
```
['navigation-bar-menu', null] // Zero index = 'navigation-bar-menu'
```
If you want to check `donate-menu` the result will be:
```
[null, 'donate-menu'] // Zero index = null
```

So, it will never get the config of the `donate-menu`

Comment (lithrel) [reply assets/src/js/menu_editor.js:160]: Yes I saw that a bit late, the fix was also part of the code I had left there https://github.com/greenpeace/planet4-master-theme/pull/1604#discussion_r810962793, works for me :+1:

**Review by mleray — APPROVED** [review]
Looks good to me 👍

**Pull request merged**
